src/Components/Server/src/Circuits/RemoteNavigationInterception.cs

```diff
@@ -35,6 +42,22 @@ public async Task EnableNavigationInterceptionAsync()
                 "attempted during prerendering or while the client is disconnected.");
         }
 
-        await _jsRuntime.InvokeAsync<object>(Interop.EnableNavigationInterception);
+        await _jsRuntime.InvokeAsync<object>(Interop.EnableNavigationInterception, _navigationManager.Uri);
+    }
+
+    public async Task DisableNavigationInterceptionAsync()
+    {
+        if (!HasAttachedJSRuntime)
+        {
+            return;
+        }
+
+        try
+        {
+            await _jsRuntime.InvokeAsync<object>(Interop.DisableNavigationInterception);
+        }
+        catch (JSDisconnectedException)
+        {
```

Comment:
Is it OK for us to consume this exception?

Reply:
Yep. This happens if the browser is no longer available, in which case there's no browser-side state to clean up. In other words, if this exception gets thrown, it means there's no work to do.